TAScheduler/Managment/LabManagement.py

```diff
@@ -1,53 +1,146 @@
-	# Side-effects: Lab is created and added inside the database
-	# Lab Name(in) - Name of the lab
-	# Lab Hours(in) - Hours of the lab
-	# Lab Location(in) - Location of the lab
-	# Lab Days(in) - Days of the lab
-	# Lab Instructor(in) - Instructor of the lab
-	# Lab TA(in) -TA of the lab
-    def createLab(self, labName, labHours, labLocation, labDays, labTA):
-        pass
 
+# class Lab(models.Model):
+#    labID = models.IntegerField(default=0)
+#    name = models.CharField(max_length=20)
+#    location = models.CharField(max_length=20)
+#    hours = models.CharField(max_length=20)
+#    days = models.CharField(max_length=20)
+#    instructor = models.ForeignKey(UserProfile, on_delete=models.PROTECT)
+#    TA = models.ForeignKey(UserProfile, on_delete=models.PROTECT, related_name="TAToLab")
 	# Preconditions: The user has to have been instantiated.
-    # The user must be of type administrator
-    # Postconditions:Edits a lab
-	# Side-effects: Lab is edited inside the database
-	# Lab Name(in) - Name of the lab
-	# Lab Hours(in) - Hours of the lab
-	# Lab Location(in) - Location of the lab
-	# Lab Days(in) - Days of the lab
-	# Lab Instructor(in) - Instructor of the lab
-	# Lab TA(in) -TA of the lab
-    def editLab(self, labName, labHours, labLocation, labDays, labTA):
-        pass
+
+    # Side-effects: Lab is created and added inside the database
+    #Lab ID(in) - Id of the lab
+    # Lab Name(in) - Name of the lab
+    # Lab Hours(in) - Hours of the lab
+    # Lab Location(in) - Location of the lab
+    # Lab Days(in) - Days of the lab
+    # Lab Instructor(in) - Instructor of the lab
+    # Lab TA(in) -TA of the lab
+    def createLab(request, Id, labName, labLocation, labHours, labDays, labInstructor, labTA):
+        if(not(isinstance(Id, int))):
+            raise TypeError("Id entered is not of type int")
+        if(not(isinstance(labName, str))):
+            raise TypeError("Lab Name entered is not of type str")
+        if(not(isinstance(labLocation, str))):
+            raise TypeError("Lab Location entered is not of type str")
+        if(not(isinstance(labHours, str))):
+            raise TypeError("Lab Hours entered is not of type str")
+        if(not(isinstance(labDays, str))):
+            raise TypeError("Lab Days entered is not of type str")
+        if(not(isinstance(labInstructor, UserProfile))):
+            raise TypeError("Lab Instructor entered is not of type User")
+        if(labInstructor.userType != "INSTRUCTOR"):
+            raise TypeError("Lab Instructor's type is not of type INSTRUCTOR")
+        if (not(isinstance(labTA, UserProfile))):
+            raise TypeError("Lab TA entered is not of type User")
+        if(labTA.userType != "TA"):
+            raise TypeError("Lab TA's type is not of type TA")
+
+        newLab = Lab.objects.create(labID=Id, name=labName, location=labLocation, 
+        hours=labHours, days=labDays, instructor=labInstructor, TA=labTA)
+        newLab.save()
```

Comment:
```suggestion
        Lab.objects.create(labID=Id, name=labName, location=labLocation, 
        hours=labHours, days=labDays, instructor=labInstructor, TA=labTA)
```
Lab.objects.create() already saves the Lab to the database so there is no need to call save.